manifests/server/grant.pp

```diff
@@ -302,18 +327,63 @@
-      false    => undef,
-      'custom' => $custom_unless,
-      default  => "SELECT 1 WHERE ${unless_function}('${role}',
-                  '${_granted_object}', '${unless_privilege}')",
+  if ($dialect == 'redshift' and $_lowercase_role =~ /^group (.*)/) {
+    # Built-in functions such as has_table_privilege don't work on
+    # groups in Redshift at this writing. Similarly,
+    # information_schema role tables do not appear to be consistently
+    # kept up to date. As such, we have to dive into the low-level
+    # aclitem[] within pg_catalog.pg_class to find what we're looking
+    # for.
+    # 
+    # This only works with object types that cleanly map to
+    # information_schema, and is incompatible with complex permission
+    # types.
+    $_lowercase_object_type = downcase($_object_type)
+    $_custom_unless = "SELECT charindex('${_privilege}', (SELECT substring(
+            case when charindex('r',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'SELECT' else '' end
+          ||case when charindex('w',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'UPDATE' else '' end
+          ||case when charindex('a',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'INSERT' else '' end
+          ||case when charindex('d',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'DELETE' else '' end
+          ||case when charindex('R',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'RULE' else '' end
+          ||case when charindex('x',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'REFERENCES' else '' end
+          ||case when charindex('t',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'TRIGGER' else '' end
+          ||case when charindex('X',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'EXECUTE' else '' end
+          ||case when charindex('U',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'USAGE' else '' end
+          ||case when charindex('C',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'CREATE' else '' end
+          ||case when charindex('T',split_part(split_part(array_to_string(relacl, '|'),${_lowercase_role},2 ) ,'/',1)) > 0 then 'TEMPORARY' else '' end
+       , 2,10000)
+    from
+    (SELECT c.relacl FROM pg_class c
+     left join pg_namespace nsp on (c.relnamespace = nsp.oid)
+    WHERE
+     nsp.nspname = '${_schema}'
+     AND c.relname = '${_relation}'
+     AND c.reltype = '${_lowercase_object_type}'
+    ))) > 0"
+    if $_lowercase_object_type == "all tables in schema" {
+      warning('UNLESS clause is not yet supported for ALL TABLES IN SCHEMA granted to groups')
```

Comment:
can you include a comment with a github issue or stackoverflow etc of why this isn't supported and/or what would be pending for it to be supported?

Reply:
https://docs.aws.amazon.com/redshift/latest/dg/c_unsupported-postgresql-functions.html describes this in detail. I'll link it in a comment.

Reply:
I'll reword this. This is being pushed to next review, due to the complexity of scanning all tables while working around https://docs.aws.amazon.com/redshift/latest/dg/c_unsupported-postgresql-functions.html.